framework/src/meshgenerators/BlockToMeshConverterGenerator.C

```diff
@@ -0,0 +1,125 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BlockToMeshConverterGenerator.h"
+#include "CastUniquePointer.h"
+#include "libmesh/elem.h"
+#include "MooseMeshUtils.h"
+
+registerMooseObject("MooseApp", BlockToMeshConverterGenerator);
+
+InputParameters
+BlockToMeshConverterGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  params.addClassDescription(
+      "Converts one or more blocks (subdomains) from a mesh into a stand-alone mesh with a "
+      "single block in it.");
+
+  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
+  params.addRequiredParam<std::vector<SubdomainName>>(
+      "target_blocks",
+      "The (list of) blocks (or 'subdomains') we wish to have moved to a new mesh (by name, not "
+      "ID)");
+
+  return params;
+}
+
+BlockToMeshConverterGenerator::BlockToMeshConverterGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _input(getMesh("input")),
+    _target_blocks(getParam<std::vector<SubdomainName>>("target_blocks"))
+{
+}
+
+std::unique_ptr<MeshBase>
+BlockToMeshConverterGenerator::generate()
+{
+  std::unique_ptr<MeshBase> mesh = std::move(_input);
+
+  if (!mesh->is_replicated())
+    mooseError("BlockToMeshConverterGenerator is not implemented for distributed meshes.");
```

Comment:
This support would be really easy to add.  I see that this error message is a common cut-and-paste-and-edit in MOOSE (`TiledMeshGenerator` forgot to change the name in the string, even!), and indeed distributed mesh support in generators can be extremely tricky, but in this case most of the algorithm should be embarrassingly parallel and the only catch would be the need to check for and null out obsolete `remote_elem` links at the end.

Would anyone want this support?

Reply:
the use cases for this are sideset extrusion and triangulator.
neither of these support distributed mesh currently though I could absolutely see it in the future. 

I dont want it right now tbh, I'm trying to have all my summer intern PRs merged at this time.